pkg/provisioner/ironic/ironic.go

```diff
@@ -1295,3 +1323,28 @@ func (p *ironicProvisioner) PowerOff() (result provisioner.Result, err error) {
 
 	return result, nil
 }
+
+// getImageChecksum parses the checksum and returns image_hash and image_value
+// e.g. "md5:458950b8cbebad6d437bd35d4d2bf66a" returns "md5" and "458950b8cbebad6d437bd35d4d2bf66a"
+// "458950b8cbebad6d437bd35d4d2bf66a" returns a default algo: "md5", and "458950b8cbebad6d437bd35d4d2bf66a"
+// "sha256:http://172.16.1.100/images/myOSv1/myOS.qcow2.sha256sum" returns "sha256" and "http://172.16.1.100/images/myOSv1/myOS.qcow2.sha256sum"
+// "http://172.16.1.100/images/myOSv1/myOS.qcow2.md5sum" returns "md5" and "http://172.16.1.100/images/myOSv1/myOS.qcow2.md5sum"
+
+func getImageChecksum(checksum string) (string, string) {
+	if checksum != "" {
+		// Check to see if the string is an URL. If so, just return
+		// the value presented
+		uri, err := url.Parse(checksum)
+		if err == nil && uri.Scheme != "" && uri.Host != "" {
```

Comment:
It's not obvious that this does the Right Thing in the case where data is of the form: `sha256:http://...` - obviously it works because there's a test, but mostly by coincidence.
If we had a list of allowable algorithms then we could just check for those prefixes.